src/csm_testing/tests/sat_functional/bootprep/test_bootprep.py

```
@@ -663,6 +664,181 @@ def delete_all_session_templates_matching_prefix(cls):
+
+        except Exception as cleanup_err:
+            logging.warning('Error during failed bootprep cleanup: %s', cleanup_err)
+
+    def _is_valid_ims_job(self, job_id):
+        """Check if a given UUID is a valid IMS job ID.
+
+        Args:
+            job_id (str): potential IMS job ID
+
+        Returns:
+            bool: True if job_id is a valid IMS job, False otherwise
+        """
+        try:
+            check_command = f'cray ims jobs describe {job_id} --format json'
+            result = subprocess.run(shlex.split(check_command),
+                                    stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            return result.returncode == 0
+        except Exception:
+            return False
+
+    def _cleanup_cfs_sessions_from_failed_bootprep(self, stderr_text):
+        """Clean up CFS sessions and their associated IMS jobs from failed bootprep.
+
+        Args:
+            stderr_text (str): stderr output from failed bootprep command
+        """
+        try:
+            # Look for CFS session names in error output
+            # CFS sessions created by bootprep typically include the configuration name
+            session_pattern = rf'{re.escape(self.test_prefix)}[\w-]*-[0-9a-f]{{8}}'
+            potential_sessions = re.findall(session_pattern, stderr_text, re.IGNORECASE)
+
+            for session_name in potential_sessions:
+                if self._is_valid_cfs_session(session_name):
+                    logging.info('Found CFS session %s in failed bootprep output, attempting cleanup', session_name)
+                    # Get associated IMS job from CFS session before deleting session
+                    ims_job_id = self._get_ims_job_from_cfs_session(session_name)
+                    if ims_job_id:
+                        self.delete_ims_job(ims_job_id)
+                    self._delete_cfs_session(session_name)
+
+        except Exception as cleanup_err:
```

This is a very broad exception clause that could mask errors indicating that we have made a coding mistake.